pkg/controller/sync/status/status.go

```diff
@@ -71,8 +73,10 @@ const (
 )
 
 type GenericClusterStatus struct {
-	Name   string            `json:"name"`
-	Status PropagationStatus `json:"status,omitempty"`
+	Name         string            `json:"name"`
+	Status       PropagationStatus `json:"status,omitempty"`
+	RemoteStatus interface{}       `json:"remoteStatus,omitempty"`
```

Comment:
not a fan of unstructured status... this answers my earlier question but needs to be defined clearly in the documentation.

Reply:
Due to the uncertain nature of the status of the different resources, we have to use this unstructured status. That is explained in the KEP.

Reply:
I would recommend using `map[string]interface{}`, which is the format of the unstructured schema.